homeassistant/components/sensor/zamg.py

```diff
@@ -0,0 +1,255 @@
+    I am a weather wrapper for a specific station and a specific attribute.
+
+    Multiple instances (one for each condition) will refer to the same
+    probe, so things will only get fetched once.
+    """
+
+    attribution = ATTRIBUTION
+    temperature_unit = TEMP_CELSIUS
+
+    def __init__(self, probe, variable, name):
+        """Init condition sensor."""
+        self.probe = probe
+        self.client_name = name
+        self.variable = variable
+
+    def update(self):
+        """Delegate update to probe."""
+        self.probe.update()
+
+    @property
+    def name(self):
+        """Build name of sensor."""
+        return "%s_%s" % (self.client_name, self.variable)
+
+    @property
+    def state(self):
+        """Return state."""
+        return self.probe.get_data(self.variable)
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes."""
+        return {
+            ATTR_WEATHER_ATTRIBUTION: self.attribution,
+            "friendly_name": SENSOR_TYPES[self.variable][0],
+            "station": self.probe.get_data('station_name'),
+            "updated": "%s %s" % (self.probe.get_data('update_date'),
+                                  self.probe.get_data('update_time'))
+
+        }
+
+
+# ------------------------------------------------------------------------
+class ZamgData(object):
+    """
+    I represent weather data for a specific site.
+
+    From the web site:
+
+    Sie beinhalten neben Stationsnummer, Stationsname, Seehöhe der Station,
+    Messdatum und Messzeit (Lokalzeit) die meteorologischen Messwerte von
+    Temperatur, Taupunkt, relative Luftfeuchtigkeit, Richtung und
+    Geschwindigkeit des Windmittels und der Windspitze, Niederschlagssumme
+    der letzten Stunde, Luftdruck reduziert auf Meeresniveau und Luftdruck
+    auf Stationsniveau sowie die Sonnenscheindauer der letzten Stunde (in
+    Prozent). Die Messstationen, die diese Daten liefern, sind über das
+    Bundesgebiet verteilt und beinhalten alle Landeshauptstädte sowie
+    die wichtigsten Bergstationen.
+    """
+
+    API_URL = "http://www.zamg.ac.at/ogd/"
+
+    API_FIELDS = {
+        v[2]: (k, v[3])
+        for k, v in SENSOR_TYPES.items()
+    }
+
+    API_HEADERS = {
+        'User-Agent': 'home-assistant.zamg/' + __version__,
+    }
+
+    def __init__(self, logger, station_id):
+        """Initialize the probe."""
+        self._logger = logger
+        self._station_id = station_id
+        self.data = {}
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
+        """
+        Update data set.
+
+        Fetch a new data set from the zamg server, parse it and
+        update internal state accordingly
+        """
+        response = requests.get(self.API_URL,
+                                headers=self.API_HEADERS, timeout=15)
+        response.encoding = 'UTF8'
+        if response.status_code == 200:
+            content_type = response.headers.get('Content-Type', 'whatever')
```

Comment:
This looks like a hack for the condition on Line 231. You could benefit of a guard clause here. Less indentation and simple code.